### PR TITLE
Admin DataGrid: Keep search terms after page reload

### PR DESCRIPTION
**Pull request by AnnikaWolff**
When searching in a DataGrid in the admin panel, the search term appears as a "post-it" beneath the search bar. 
image

But as soon as a filter is added, e.g. a channel is selected, the search term disappears, although it remains as a URL parameter. The result is filtered correctly, but the user is confused, because the search term is not shown.

This PR fixes this issue: The search term will remain with all applied filters beneath the search bar. The channel and the customer group will appear there as well now.
That means, the gray terms are directly corresponding to the URL parameters.
image

EDIT: I just added encoding of the search terms, to let us be able to search for weird coupon codes as "AB#CD" or "All20%Off".

### CODE REVIEW

File: packages/Webkul/Ui/src/DataGrid/DataGrid.php [hunk 1]
```diff
@@ -400,12 +400,12 @@ public function sortOrFilterCollection($collection, $parseInfo)
                 }
             } else {
                 foreach ($this->completeColumnDetails as $column) {
-                    if($column['index'] == $columnName && !$column['filterable']) {
+                    if($column['index'] === $columnName && !$column['filterable']) {
```
Comment (jitendra-webkul) [inline packages/Webkul/Ui/src/DataGrid/DataGrid.php:403]: Add space after if

File: packages/Webkul/Ui/src/Resources/views/datagrid/table.blade.php [hunk 2]
```diff
@@ -290,51 +290,52 @@
                     getResponse: function() {
                         label = '';
 
-                        for(colIndex in this.columns) {
+                        for(let colIndex in this.columns) {
                             if(this.columns[colIndex].index == this.columnOrAlias) {
```
Comment (jitendra-webkul) [inline packages/Webkul/Ui/src/Resources/views/datagrid/table.blade.php:294]: Add space after if

File: packages/Webkul/Ui/src/Resources/views/datagrid/table.blade.php [hunk 3]
```diff
@@ -290,51 +290,52 @@
...
 
-                            if (this.filterIndex == this.columnOrAlias && (this.numberValue == 0 || this.numberValue < 0)) {
+                            if (this.filterIndex === this.columnOrAlias
+                                && (this.numberValue === 0 || this.numberValue < 0)) {
                                 indexConditions = false;
 
                                 alert('{{__('ui::app.datagrid.zero-index')}}');
                             }
 
                             if(indexConditions)
```
Comment (jitendra-webkul) [inline packages/Webkul/Ui/src/Resources/views/datagrid/table.blade.php:312]: Add space after if

File: packages/Webkul/Ui/src/Resources/views/datagrid/table.blade.php [hunk 4]
```diff
@@ -290,51 +290,52 @@
...
                         }
                     },
 
                     sortCollection: function(alias) {
-                        label = '';
+                        let label = '';
 
-                        for(colIndex in this.columns) {
-                            if(this.columns[colIndex].index == alias) {
+                        for(let colIndex in this.columns) {
```
Comment (jitendra-webkul) [inline packages/Webkul/Ui/src/Resources/views/datagrid/table.blade.php:326]: Please add space after for

File: packages/Webkul/Ui/src/Resources/views/datagrid/table.blade.php [hunk 5]
```diff
@@ -345,7 +346,7 @@
                             this.arrayFromUrl();
                         }
 
-                        for(id in this.massActions) {
+                        for(let id in this.massActions) {
```
Comment (jitendra-webkul) [inline packages/Webkul/Ui/src/Resources/views/datagrid/table.blade.php:349]: Please add space after for

File: packages/Webkul/Ui/src/Resources/views/datagrid/table.blade.php [hunk 6]
```diff
@@ -355,34 +356,34 @@
...
+                            if (this.massActions[id].type === 'update') {
                                 this.massActionValues = this.massActions[id].options;
                             }
                         }
                     },
 
                     findCurrentSort: function() {
-                        for(i in this.filters) {
-                            if (this.filters[i].column == 'sort') {
+                        for(let i in this.filters) {
```
Comment (jitendra-webkul) [inline packages/Webkul/Ui/src/Resources/views/datagrid/table.blade.php:366]: Please add space after for

File: packages/Webkul/Ui/src/Resources/views/datagrid/table.blade.php [hunk 7]
```diff
@@ -355,34 +356,34 @@
...
                             }
                         }
                     },
 
                     changeMassActionTarget: function() {
-                        if (this.massActionType == 'delete') {
-                            for(i in this.massActionTargets) {
-                                if (this.massActionTargets[i].type == 'delete') {
+                        if (this.massActionType === 'delete') {
+                            for(let i in this.massActionTargets) {
```
Comment (jitendra-webkul) [inline packages/Webkul/Ui/src/Resources/views/datagrid/table.blade.php:375]: √

Comment (jitendra-webkul) [inline packages/Webkul/Ui/src/Resources/views/datagrid/table.blade.php:375]: Please add space after for

File: packages/Webkul/Ui/src/Resources/views/datagrid/table.blade.php [hunk 8]
```diff
@@ -355,34 +356,34 @@
...
                                     break;
                                 }
                             }
                         }
 
-                        if (this.massActionType == 'update') {
-                            for(i in this.massActionTargets) {
-                                if (this.massActionTargets[i].type == 'update') {
+                        if (this.massActionType === 'update') {
+                            for(let i in this.massActionTargets) {
```
Comment (jitendra-webkul) [inline packages/Webkul/Ui/src/Resources/views/datagrid/table.blade.php:385]: Please add space after for

File: packages/Webkul/Ui/src/Resources/views/datagrid/table.blade.php [hunk 9]
```diff
@@ -397,23 +398,26 @@
...
-                                if (column != "sort" && column != "search") {
-                                    filterRepeated = 0;
+                                if (column !== "sort" && column !== "search") {
+                                    let filterRepeated = false;
 
-                                    for(j = 0; j < this.filters.length; j++) {
-                                        if (this.filters[j].column == column) {
-                                            if (this.filters[j].cond == condition && this.filters[j].val == response) {
-                                                filterRepeated = 1;
+                                    for(let j = 0; j < this.filters.length; j++) {
```
Comment (jitendra-webkul) [inline packages/Webkul/Ui/src/Resources/views/datagrid/table.blade.php:413]: Please add space after for

File: packages/Webkul/Ui/src/DataGrid/DataGrid.php (hunk 1 above)
Comment (AnnikaWolff) [reply packages/Webkul/Ui/src/DataGrid/DataGrid.php:403]: Done.

File: packages/Webkul/Ui/src/Resources/views/datagrid/table.blade.php (hunk 2 above)
Comment (AnnikaWolff) [reply packages/Webkul/Ui/src/Resources/views/datagrid/table.blade.php:294]: Done.

File: packages/Webkul/Ui/src/Resources/views/datagrid/table.blade.php (hunk 3 above)
Comment (AnnikaWolff) [reply packages/Webkul/Ui/src/Resources/views/datagrid/table.blade.php:312]: Done

File: packages/Webkul/Ui/src/Resources/views/datagrid/table.blade.php (hunk 4 above)
Comment (AnnikaWolff) [reply packages/Webkul/Ui/src/Resources/views/datagrid/table.blade.php:326]: Done.

File: packages/Webkul/Ui/src/Resources/views/datagrid/table.blade.php (hunk 5 above)
Comment (AnnikaWolff) [reply packages/Webkul/Ui/src/Resources/views/datagrid/table.blade.php:349]: Done.

File: packages/Webkul/Ui/src/Resources/views/datagrid/table.blade.php (hunk 7 above)
Comment (AnnikaWolff) [reply packages/Webkul/Ui/src/Resources/views/datagrid/table.blade.php:375]: Done.

File: packages/Webkul/Ui/src/Resources/views/datagrid/table.blade.php (hunk 8 above)
Comment (AnnikaWolff) [reply packages/Webkul/Ui/src/Resources/views/datagrid/table.blade.php:385]: Done.

File: packages/Webkul/Ui/src/Resources/views/datagrid/table.blade.php (hunk 9 above)
Comment (AnnikaWolff) [reply packages/Webkul/Ui/src/Resources/views/datagrid/table.blade.php:413]: Done. 

And found some more on my way through the file.
Thanks for reviewing :)

**Comment by vaishaliwebkul** [comment]
Hi @AnnikaWolff 
Please correct this
Screenshot_from_2020_06_18_12_56_26

**Comment by AnnikaWolff** [comment]
Hi @vaishaliwebkul 

Yes, the search term is converted from a url encoded string to a url decoded string for the sake of better readability.
But I do not think, that is a problem, do you?

For instance, I have a simple product called "Simple Product". I search for it with the terms "Simple Product", having a space between the two words. In the URL, the search terms are encoded to "Simple%20Product", that is not nice to read. So coming out of the URL, the terms are decoded to "Simple+Product", which is easier to read. 

The search result is always the same: If looking for "Simple Product", "Simple%20Product" or "Simple+Product", I always find my simple product.

Concerning the failed check, I already notified @jitendra-webkul about the faulty migration causing those errors.

**Comment by vaishaliwebkul** [comment]
@AnnikaWolff 

1. My concern is for search key changed when locale added; 
for example When I search for 'simple product' the search result be shown 'simple product', as and when I added the locale key 'en' then previous search key will be decoded to 'simple+product', I think it's a bug, we should keep the search key same before and after adding locale key.

2. Another Bug I just found while filtering list on the basis of status->True/Active (converted to False/Inactive)

Screenshot (7)

**Comment by AnnikaWolff** [comment]
Hi @vaishaliwebkul 

Alright, I will check that.

**Comment by AnnikaWolff** [comment]
Dear @vaishaliwebkul 

1st: I removed the "+" symbol that appeared when searching for multiple terms and adding a locale filter.
2nd: I fixed the bug of the boolean filter always showing the "false / inactive" value.

And 3rd: I added a label to the locale filter.

image

Please feel free to test again.

**Comment by vaishaliwebkul** [comment]
Hi @AnnikaWolff 
Thanks for resolving the above issue :)

One more update, In the cart rule page, filtered channels & customer groups tag with their ID, can't we show them by their name likewise in the product list.

Screenshot (26)


Screenshot (26)

**Comment by AnnikaWolff** [comment]
Hi @vaishaliwebkul 

This is much more effort because I have to refactor all DataGrids and their views at once. 
So stay tuned, it may take a while.

**Comment by AnnikaWolff** [comment]
Hi  @vaishaliwebkul 

Done :) 
Filtered channels, locales and customer groups are now shown by their names instead of code or id.

And all my changes have another advantage, I suppose: It is much easier now to add those filters to other DataGrids, if needed.

**Pull request merged**